source/MaterialXContrib/MaterialXNode/MaterialXData.h

```diff
@@ -22,13 +24,29 @@ struct MaterialXData
     MaterialXData& operator=(const MaterialXData&) = delete;
     MaterialXData& operator=(MaterialXData&&) = delete;
 
-	MaterialX::FilePath libSearchPath;
-	MaterialX::DocumentPtr doc;
-	MaterialX::ElementPtr element;
-
-	MString fragmentName;
-	std::unique_ptr<MaterialX::OGSXMLFragmentWrapper> glslFragmentWrapper;
-	std::vector<std::unique_ptr<MaterialX::GenContext>> contexts;
+    MaterialX::DocumentPtr getDocument() const
+    {
+        return _doc;
+    }
+
+    const MString& getFragmentName()
+    {
+        return _fragmentName;
+    }
+
+    MaterialX::OGSXMLFragmentWrapper* getFragmentWrapper() const
+    {
+        return _xmlFragmentWrapper.get();
+    }
+
+  protected:
+	MaterialX::FilePath _libSearchPath;
+	MaterialX::DocumentPtr _doc;
+	MaterialX::ElementPtr _element;
+
+	MString _fragmentName;
+    OGSXMLFragmentWrapperPtr _xmlFragmentWrapper;
```

Comment:
@Pavlo, I think you still have the incorrect formatting settings. Tabs as spaces?

Reply:
I do have tabs to spaces set up. That leaves @feldstj :-)